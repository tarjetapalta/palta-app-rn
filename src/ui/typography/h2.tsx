import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const H2: FunctionComponent<
  Pick<TextProps, 'testID' | 'color' | 'emphasis' | 'textAlign'>
> = ({
  /** Test id for test */
  testID = 'H2',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="xl"
    weight="bold"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
