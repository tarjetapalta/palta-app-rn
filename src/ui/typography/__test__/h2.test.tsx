import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { H2 } from '../h2';

describe('h2 component', () => {
  it('basic H2 with color "text", emphasis "high" and testID "H2"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H2>Some text</H2>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic H2 with color "primary", emphasis "medium"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H2 color="primary" emphasis="medium">
            Some text
          </H2>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic with custom testId', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H2 testID="custom testID">Some text</H2>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
