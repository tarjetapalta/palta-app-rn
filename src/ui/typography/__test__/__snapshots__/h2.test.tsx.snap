// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h2 component basic H2 with color "primary", emphasis "medium" 1`] = `
<Text
  color="primary"
  emphasis="medium"
  numberOfLines={0}
  size="xl"
  style={
    Array [
      Object {
        "color": "#91b086",
        "fontFamily": "BwModelica-Bold",
        "fontSize": 24,
        "opacity": 0.6,
        "textAlign": "center",
      },
    ]
  }
  testID="H2"
  weight="bold"
>
  Some text
</Text>
`;

exports[`h2 component basic H2 with color "text", emphasis "high" and testID "H2" 1`] = `
<Text
  numberOfLines={0}
  size="xl"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Bold",
        "fontSize": 24,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="H2"
  weight="bold"
>
  Some text
</Text>
`;

exports[`h2 component basic with custom testId 1`] = `
<Text
  numberOfLines={0}
  size="xl"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Bold",
        "fontSize": 24,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="custom testID"
  weight="bold"
>
  Some text
</Text>
`;
